test/TestApps/ToastNotificationsTestApp/main.cpp

```diff
@@ -435,6 +458,189 @@ std::string unitTestNameFromLaunchArguments(const winrt::ILaunchActivatedEventAr
     return unitTestName;
 }
 
+winrt::ToastProgressResult ProgressResultOperationHelper(winrt::IAsyncOperation<winrt::ToastProgressResult> progressResultOperation)
+{
+    if (progressResultOperation.wait_for(std::chrono::seconds(2)) != winrt::AsyncStatus::Completed)
+    {
+        progressResultOperation.Cancel();
+        return winrt::ToastProgressResult::Failed; // timed out or failed
+    }
+
+    return progressResultOperation.GetResults();
+}
+
+bool PostToastHelper(std::wstring const& tag, std::wstring const& group)
+{
+    winrt::ToastNotification toast{ GetToastNotification() };
+
+    toast.Tag(tag.c_str());
+    toast.Group(group.c_str());
+
+    winrt::ToastNotificationManager::Default().ShowToast(toast);
+
+    if (toast.ToastId() == 0)
+    {
+        return false;
+    }
+
+    return true;
+}
+
+bool VerifyUpdateToastProgressDataUsingValidTagAndValidGroup()
+{
+    // Registration happens in main()
+    PostToastHelper(L"PTag", L"PGroup");
+
+    winrt::ToastProgressData progressData = GetToastProgressData(L"PStatus", L"PTitle", 0.10, L"10%");
+
+    auto progressResultOperation = winrt::ToastNotificationManager::Default().UpdateToastProgressDataAsync(progressData, L"PTag", L"PGroup");
+    auto progressResult = ProgressResultOperationHelper(progressResultOperation);
+
+    return progressResult == winrt::ToastProgressResult::Succeeded;
+}
+
+bool VerifyUpdateToastProgressDataUsingValidTagAndValidGroup_Unpackaged()
+{
+    winrt::ToastAssets assets(L"ToastNotificationApp", winrt::Uri{ LR"(C:\Windows\System32\WindowsSecurityIcon.png)" });
+    auto activationInfo = winrt::ToastActivationInfo::CreateFromToastAssets(assets);
+
+    winrt::ToastNotificationManager::Default().RegisterActivator(activationInfo);
```

Comment:
It's a little inefficient keep calling `Default()` again and again. Make it a local variable.

Reply:
Please address it to all applicable tests.